frontend/src/buttons/AddUserButton/AddUserButton.tsx

```diff
@@ -3,11 +3,13 @@ import { useLocation, useHistory } from 'react-router-dom'
 import { IconButton, Tooltip } from '@material-ui/core'
 import { Icon } from '../../components/Icon'
 
-export const AddUserButton: React.FC = () => {
+export const AddUserButton: React.FC<{ device?: IDevice }> = ({ device }) => {
   const location = useLocation()
   const history = useHistory()
   const onClick = () => history.push(`${location.pathname.replace('/users', '')}/users/share`)
 
+  if (device?.shared) return null
```

Comment:
I like to put the component visibility in each component, so it's portable and self contained.